contracts/token/ERC721/extensions/ERC721Pausable.sol

```diff
@@ -35,6 +40,8 @@ abstract contract ERC721Pausable is ERC721, Pausable {
     ) internal virtual override {
         super._beforeTokenTransfer(from, to, firstTokenId, batchSize);
 
-        require(!paused(), "ERC721Pausable: token transfer while paused");
+        if (paused()) {
+            revert ERC721Paused();
```

Comment:
Same as ERC20Pausable and ERC1155Pausable

Reply:
I used `_requireNotPaused` here instead to not alter the order of execution, but I don't see why it could be bad, do you think it's better to use the `whenNotPaused` modifier here?

Reply:
we'll remove the hook and go with `_update` in ERC721 at some point